flax/linen/module.py

```diff
@@ -1033,6 +1033,14 @@ def make_rng(self, name: str) -> PRNGKey:
       raise ValueError("Can't use RNGs on unbound modules")
     return self.scope.make_rng(name)
 
+  def is_under_init(self) -> bool:
+    if self.scope is None:
+      raise ValueError("Can't check if under init on unbound modules")
+    if self.scope.flags and 'initializing' in self.scope.flags:
```

Comment:
I think it would be less error-prone if flags have no defaults. Here you have default behavior and typos default to False. It would be more robust to define flags as a dict without the options to define defaults at the call site. So if a user requires self.is_initializing the flag **must** be set